utils/qtrunefu/qtdatagen.py

```diff
@@ -0,0 +1,183 @@
+    def __init__(self, configuration, directories):
+        super(QDialog, self).__init__()
+        self.dirs = directories
+        self.cfg = configuration
+        self.create_layout()
+
+        buttonBox = QDialogButtonBox(QDialogButtonBox.Ok | QDialogButtonBox.Cancel)
+        buttonBox.accepted.connect(self.on_accepted)
+        buttonBox.rejected.connect(self.reject)
+
+        mainLayout = QVBoxLayout()
+        mainLayout.addWidget(self.config_group_box)
+        mainLayout.addWidget(self.files_group_box)
+        mainLayout.addWidget(self.options_box)
+        mainLayout.addWidget(buttonBox)
+        self.setLayout(mainLayout)
+
+        self.setWindowTitle("Launch Event Formation Unit")
+
+    # Add a 'row' consisting of a label and a text box widget
+    def add_row(self, layout, label, type):
+        lbl = QLabel(label)
+        type.setMinimumWidth(600)
+        layout.addRow(lbl, type)
+
+    def create_layout(self):
+        self.config_group_box = QGroupBox("Directories")
+        toplayout = QFormLayout()
+
+        self.basedirle = QLineEdit(self.dirs.basedir)
+        self.add_row(toplayout, "basedir:", self.basedirle)
+        self.efudirle = QLineEdit(self.dirs.efudir)
+        self.add_row(toplayout, "efu dir:", self.efudirle)
+        self.datadirle = QLineEdit(self.dirs.datadir)
+        self.add_row(toplayout, "data dir:", self.datadirle)
+        updateb = QPushButton("Reload")
+        updateb.setMaximumWidth(100)
+        updateb.clicked.connect(self.update)
+        toplayout.addRow(updateb)
+        self.config_group_box.setLayout(toplayout)
+
+        self.files_group_box = QGroupBox("Select configuration")
+        fileslayout = QFormLayout()
+        self.gencb = QComboBox()
+        self.add_row(fileslayout, "EFU:", self.gencb)
+        self.datacb = QComboBox()
+        self.add_row(fileslayout, "Detector:", self.datacb)
+        self.files_group_box.setLayout(fileslayout)
+
+        self.options_box = QGroupBox("Options")
+        optslayout = QFormLayout()
+        self.efuiple = QLineEdit(cfg.options['efu'])
+        self.add_row(optslayout, "EFU:", self.efuiple)
+        self.throttlele = QLineEdit(cfg.options['throttle'])
+        self.add_row(optslayout, "throttle:", self.throttlele)
+        self.options_box.setLayout(optslayout)
+
+    def _populate_field(self, field, list):
+        for name in list:
+            field.addItem(name)
+
+    def populate(self, e, d):
+        self._populate_field(self.gencb, e)
+        self._populate_field(self.datacb, d)
+
+    def get_selection(self):
+        return self._gen, self._data, self._efuip, self._throttle
+
+    # Override builtin on_accepted method for pressing OK button
+    def on_accepted(self):
+        self._gen = self.gencb.itemText(self.gencb.currentIndex())
+        self._data = self.datacb.itemText(self.datacb.currentIndex())
+        self._efuip = self.efuiple.text()
+        self._throttle = self.throttlele.text()
```

Comment:
tja, jeg er ikke 100% vild med at det hele er lowercase ud i et. Det er self din stil, men jeg tror hurtigt at det kan vaere tricky, med laengere sammensaetninger. Vil camelCase ikke vaere ok? self.throttleLe, self.efuIpLe ?

Reply:
I will check with Matt Clarke, but I believe this is more or less the python coding style. If not I will make the necessary changes.

Reply:
Matt replied:
PEP8 is the Python guidelines. https://pep8.org/
It is pretty much universally used.
In short, variable and function names are snake-case, e.g. detid or det_id, convert_to_tof() etc.
For the other aspects like spacing, indentation, etc. we tend to use the Black tool (https://pypi.org/project/black/) to automate that to conform to PEP8